include/zephyr/bluetooth/mesh/foundation.h

```diff
@@ -0,0 +1,84 @@
+/** @file
+ *  @brief Foundation model helper APIs.
+ */
+
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_MESH_FOUNDATION_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_MESH_FOUNDATION_H_
+
+#include <zephyr/sys/byteorder.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Check the number of packed key index entries in a buffer.
+ *
+ * @param buf Message buffer.
+ *
+ * @return Number of key index entries in the buffer, 0 if the size
+ * of the buffer is an invalid size.
+ */
+static inline size_t key_idx_cnt_in_buf(struct net_buf_simple *buf)
+{
+	if (!buf->len || (buf->len % 3) % 2) {
+		return 0;
+	}
+
+	return (buf->len - (buf->len % 3)) / 3 * 2 + ((buf->len % 3) ? 1 : 0);
+}
+
+/** @brief Unpack a pair of key index entries from a buffer.
+ *
+ * @param buf  Message buffer.
+ * @param idx1 Pointer to the first return index.
+ * @param idx2 Pointer to the second return index.
+ */
+static inline void key_idx_unpack_pair(struct net_buf_simple *buf, uint16_t *idx1, uint16_t *idx2)
+{
+	*idx1 = sys_get_le16(&buf->data[0]) & 0xfff;
+	*idx2 = sys_get_le16(&buf->data[1]) >> 4;
+	net_buf_simple_pull(buf, 3);
+}
+
+/** @brief Unpack a list of key index entries from a buffer.
+ *
+ * @param buf Message buffer.
+ * @param dst_arr Destination array for the unpacked list.
+ * @param dst_cnt Size of the destination array.
+ *
+ * @return Number of parsed entries in the buffer.
+ * @return -EMSGSIZE if dst_arr size is to small to parse full message.
+ */
+static inline int key_idx_unpack_list(struct net_buf_simple *buf, uint16_t *dst_arr,
```

Comment:
The complex functions shouldn't be `inline`.  Inlining huge functions will make the final binary bigger without a visible speed advantage. 
The reason to add `inline` modification for small functions is comparable complexity of function and function's prologue\epilogue. If the function's body is huge enough then inlining doesn't have a lot of sense. 
I suggest making `key_idx_unpack_list` and `key_idx_pack_list`as a normal functions.